tests/appsec/test_traces.py

```diff
@@ -293,24 +293,31 @@ def validate_response_headers(span):
 @scenarios.external_processing
 @scenarios.default
 class Test_CollectDefaultRequestHeader:
-    HEADERS = ["User-Agent", "Accept", "Content-Type"]
+    HEADERS = {
+        "User-Agent": "MyBrowser",
+        "Accept": "*/*",
+        "Content-Type": "text/plain",
+    }
 
     def setup_collect_default_request_headers(self):
-        self.r = weblog.get("/headers", headers={header: "myHeaderValue" for header in self.HEADERS})
+        self.r = weblog.get("/headers", headers=self.HEADERS)
 
     def test_collect_default_request_headers(self):
         """Collect User agent and other headers and other security info when appsec is enabled."""
-
-        def assert_header_in_span_meta(span, header):
-            if header not in span["meta"]:
-                raise Exception(f"Can't find {header} in span's meta")
-
-        def validate_request_headers(span):
-            for header in self.HEADERS:
-                assert_header_in_span_meta(span, f"http.request.headers.{header.lower()}")
-            return True
-
-        interfaces.library.validate_spans(self.r, validate_request_headers)
+        if context.library != "golang":
+            # TODO(APPSEC-56898): Golang weblogs do not respond to this request.
+            assert self.r.status_code == 200
```

Comment:
should that be a bug/missing_feature instead ? only thing you're testing here is the weblog endpoint exists, no feature tested